### ISSUE TITLE
Intel AX210NGW Wifi Firmware issue

### ISSUE THREAD

**Issue by robotastic-io** [home-assistant/operating-system#2283]
### Describe the issue you are experiencing

Hi,

I tried installing a Intel AX210NGW Wifi Card it my HA machine. After bootup bluetooth was detected fine but no Wifi. After checing dmesg I see the following log messages for iwlwifi.

> [    2.027870] iwlwifi 0000:02:00.0: api flags index 2 larger than supported by driver
[    2.027929] iwlwifi 0000:02:00.0: TLV_FW_FSEQ_VERSION: FSEQ Version: 0.63.2.2
[    2.029140] iwlwifi 0000:02:00.0: loaded firmware version 66.f1c864e0.0 ty-a0-gf-a0-66.ucode op_mode iwlmvm
[    2.154663] iwlwifi 0000:02:00.0: Detected Intel(R) Wi-Fi 6 AX210 160MHz, REV=0x420
[    2.306033] iwlwifi 0000:02:00.0: Detected RF GF, rfid=0x10d000
[    2.307027] iwlwifi 0000:02:00.0: Microcode SW error detected. Restarting 0x0.
[    2.307155] iwlwifi 0000:02:00.0: Start IWL Error Log Dump:
[    2.307162] iwlwifi 0000:02:00.0: Transport status: 0x0000004A, valid: 6
[    2.307169] iwlwifi 0000:02:00.0: Loaded firmware version: 66.f1c864e0.0 ty-a0-gf-a0-66.ucode
[    2.307178] iwlwifi 0000:02:00.0: 0x00000071 | NMI_INTERRUPT_UMAC_FATAL
[    2.307199] iwlwifi 0000:02:00.0: 0x002002F0 | trm_hw_status0
[    2.307206] iwlwifi 0000:02:00.0: 0x00000000 | trm_hw_status1
[    2.307213] iwlwifi 0000:02:00.0: 0x004DA722 | branchlink2
[    2.307219] iwlwifi 0000:02:00.0: 0x004D0CCE | interruptlink1
[    2.307225] iwlwifi 0000:02:00.0: 0x004D0CCE | interruptlink2
[    2.307231] iwlwifi 0000:02:00.0: 0x004D94DA | data1
[    2.307238] iwlwifi 0000:02:00.0: 0x00000010 | data2
[    2.307244] iwlwifi 0000:02:00.0: 0x00000000 | data3
[    2.307249] iwlwifi 0000:02:00.0: 0x00000000 | beacon time
[    2.307255] iwlwifi 0000:02:00.0: 0x00011AD6 | tsf low
[    2.307261] iwlwifi 0000:02:00.0: 0x00000000 | tsf hi
[    2.307267] iwlwifi 0000:02:00.0: 0x00000000 | time gp1
[    2.307273] iwlwifi 0000:02:00.0: 0x00023244 | time gp2
[    2.307279] iwlwifi 0000:02:00.0: 0x00000001 | uCode revision type
[    2.307285] iwlwifi 0000:02:00.0: 0x00000042 | uCode version major
[    2.307291] iwlwifi 0000:02:00.0: 0xF1C864E0 | uCode version minor
[    2.307298] iwlwifi 0000:02:00.0: 0x00000420 | hw version
[    2.307304] iwlwifi 0000:02:00.0: 0x18C89002 | board version
[    2.307311] iwlwifi 0000:02:00.0: 0x8008FF05 | hcmd
[    2.307317] iwlwifi 0000:02:00.0: 0x00020000 | isr0
[    2.307323] iwlwifi 0000:02:00.0: 0x60000000 | isr1
[    2.307330] iwlwifi 0000:02:00.0: 0x48F00002 | isr2
[    2.307335] iwlwifi 0000:02:00.0: 0x00C0000C | isr3
[    2.307342] iwlwifi 0000:02:00.0: 0x00000000 | isr4
[    2.307347] iwlwifi 0000:02:00.0: 0x00000000 | last cmd Id
[    2.307354] iwlwifi 0000:02:00.0: 0x004D94DA | wait_event
[    2.307360] iwlwifi 0000:02:00.0: 0x00000000 | l2p_control
[    2.307366] iwlwifi 0000:02:00.0: 0x00000000 | l2p_duration
[    2.307373] iwlwifi 0000:02:00.0: 0x00000000 | l2p_mhvalid
[    2.307379] iwlwifi 0000:02:00.0: 0x00000000 | l2p_addr_match
[    2.307385] iwlwifi 0000:02:00.0: 0x00000009 | lmpm_pmg_sel
[    2.307393] iwlwifi 0000:02:00.0: 0x00000000 | timestamp
[    2.307399] iwlwifi 0000:02:00.0: 0x00000024 | flow_handler
[    2.307452] iwlwifi 0000:02:00.0: Start IWL Error Log Dump:
[    2.307458] iwlwifi 0000:02:00.0: Transport status: 0x0000004A, valid: 7
[    2.307465] iwlwifi 0000:02:00.0: 0x2010070D | ADVANCED_SYSASSERT
[    2.307473] iwlwifi 0000:02:00.0: 0x00000000 | umac branchlink1
[    2.307479] iwlwifi 0000:02:00.0: 0x8045DFC6 | umac branchlink2
[    2.307485] iwlwifi 0000:02:00.0: 0x010910FE | umac interruptlink1
[    2.307491] iwlwifi 0000:02:00.0: 0x00000000 | umac interruptlink2
[    2.307497] iwlwifi 0000:02:00.0: 0x00000005 | umac data1
[    2.307502] iwlwifi 0000:02:00.0: 0xDEADBEEF | umac data2
[    2.307508] iwlwifi 0000:02:00.0: 0xDEADBEEF | umac data3
[    2.307514] iwlwifi 0000:02:00.0: 0x00000042 | umac major
[    2.307520] iwlwifi 0000:02:00.0: 0xF1C864E0 | umac minor
[    2.307527] iwlwifi 0000:02:00.0: 0x0002323C | frame pointer
[    2.307534] iwlwifi 0000:02:00.0: 0xC0885E88 | stack pointer
[    2.307540] iwlwifi 0000:02:00.0: 0x00010C00 | last host cmd
[    2.307546] iwlwifi 0000:02:00.0: 0x00000000 | isr status reg
[    2.307565] iwlwifi 0000:02:00.0: IML/ROM dump:
[    2.307571] iwlwifi 0000:02:00.0: 0x00000B03 | IML/ROM error/state
[    2.307590] iwlwifi 0000:02:00.0: 0x00007279 | IML/ROM data1
[    2.307608] iwlwifi 0000:02:00.0: 0x00000080 | IML/ROM WFPM_AUTH_KEY_0
[    2.307622] iwlwifi 0000:02:00.0: Fseq Registers:
[    2.307631] iwlwifi 0000:02:00.0: 0x60000100 | FSEQ_ERROR_CODE
[    2.307641] iwlwifi 0000:02:00.0: 0x00440005 | FSEQ_TOP_INIT_VERSION
[    2.307651] iwlwifi 0000:02:00.0: 0x00080009 | FSEQ_CNVIO_INIT_VERSION
[    2.307662] iwlwifi 0000:02:00.0: 0x0000A652 | FSEQ_OTP_VERSION
[    2.307672] iwlwifi 0000:02:00.0: 0x00000002 | FSEQ_TOP_CONTENT_VERSION
[    2.307683] iwlwifi 0000:02:00.0: 0x4552414E | FSEQ_ALIVE_TOKEN
[    2.307693] iwlwifi 0000:02:00.0: 0x00400410 | FSEQ_CNVI_ID
[    2.307703] iwlwifi 0000:02:00.0: 0x00400410 | FSEQ_CNVR_ID
[    2.307713] iwlwifi 0000:02:00.0: 0x00400410 | CNVI_AUX_MISC_CHIP
[    2.307726] iwlwifi 0000:02:00.0: 0x00400410 | CNVR_AUX_MISC_CHIP
[    2.307738] iwlwifi 0000:02:00.0: 0x00009061 | CNVR_SCU_SD_REGS_SD_REG_DIG_DCDC_VTRIM
[    2.307751] iwlwifi 0000:02:00.0: 0x00000061 | CNVR_SCU_SD_REGS_SD_REG_ACTIVE_VDIG_MIRROR
[    2.307776] iwlwifi 0000:02:00.0: WRT: Collecting data: ini trigger 13 fired (delay=0ms).
[    3.345305] iwlwifi 0000:02:00.0: Failed to run INIT ucode: -5
[    3.357230] iwlwifi 0000:02:00.0: retry init count 0
[    3.357764] iwlwifi 0000:02:00.0: Detected Intel(R) Wi-Fi 6 AX210 160MHz, REV=0x420
[    3.509885] iwlwifi 0000:02:00.0: Microcode SW error detected. Restarting 0x0.
[    3.510024] iwlwifi 0000:02:00.0: Start IWL Error Log Dump:
[    3.510032] iwlwifi 0000:02:00.0: Transport status: 0x0000004A, valid: 6
[    3.510036] iwlwifi 0000:02:00.0: Loaded firmware version: 66.f1c864e0.0 ty-a0-gf-a0-66.ucode
[    3.510040] iwlwifi 0000:02:00.0: 0x00000071 | NMI_INTERRUPT_UMAC_FATAL
[    3.510060] iwlwifi 0000:02:00.0: 0x002002F0 | trm_hw_status0
[    3.510063] iwlwifi 0000:02:00.0: 0x00000000 | trm_hw_status1
[    3.510066] iwlwifi 0000:02:00.0: 0x004DA722 | branchlink2
[    3.510069] iwlwifi 0000:02:00.0: 0x004D0CCE | interruptlink1
[    3.510087] iwlwifi 0000:02:00.0: 0x004D0CCE | interruptlink2
[    3.510090] iwlwifi 0000:02:00.0: 0x004D94DA | data1
[    3.510093] iwlwifi 0000:02:00.0: 0x00000010 | data2
[    3.510096] iwlwifi 0000:02:00.0: 0x00000000 | data3
[    3.510099] iwlwifi 0000:02:00.0: 0x00000000 | beacon time
[    3.510102] iwlwifi 0000:02:00.0: 0x00011AAC | tsf low
[    3.510121] iwlwifi 0000:02:00.0: 0x00000000 | tsf hi
[    3.510124] iwlwifi 0000:02:00.0: 0x00000000 | time gp1
[    3.510127] iwlwifi 0000:02:00.0: 0x0002321C | time gp2
[    3.510130] iwlwifi 0000:02:00.0: 0x00000001 | uCode revision type
[    3.510153] iwlwifi 0000:02:00.0: 0x00000042 | uCode version major
[    3.510157] iwlwifi 0000:02:00.0: 0xF1C864E0 | uCode version minor
[    3.510160] iwlwifi 0000:02:00.0: 0x00000420 | hw version
[    3.510162] iwlwifi 0000:02:00.0: 0x18C89002 | board version
[    3.510177] iwlwifi 0000:02:00.0: 0x8008FF05 | hcmd
[    3.510180] iwlwifi 0000:02:00.0: 0x00020000 | isr0
[    3.510204] iwlwifi 0000:02:00.0: 0x20000000 | isr1
[    3.510208] iwlwifi 0000:02:00.0: 0x48F00002 | isr2
[    3.510210] iwlwifi 0000:02:00.0: 0x00C0000C | isr3
[    3.510213] iwlwifi 0000:02:00.0: 0x00000000 | isr4
[    3.510230] iwlwifi 0000:02:00.0: 0x00000000 | last cmd Id
[    3.510236] iwlwifi 0000:02:00.0: 0x004D94DA | wait_event
[    3.510242] iwlwifi 0000:02:00.0: 0x00000000 | l2p_control
[    3.510248] iwlwifi 0000:02:00.0: 0x00000000 | l2p_duration
[    3.510253] iwlwifi 0000:02:00.0: 0x00000000 | l2p_mhvalid
[    3.510259] iwlwifi 0000:02:00.0: 0x00000000 | l2p_addr_match
[    3.510265] iwlwifi 0000:02:00.0: 0x00000009 | lmpm_pmg_sel
[    3.510271] iwlwifi 0000:02:00.0: 0x00000000 | timestamp
[    3.510276] iwlwifi 0000:02:00.0: 0x00000024 | flow_handler
[    3.510328] iwlwifi 0000:02:00.0: Start IWL Error Log Dump:
[    3.510333] iwlwifi 0000:02:00.0: Transport status: 0x0000004A, valid: 7
[    3.510341] iwlwifi 0000:02:00.0: 0x2010070D | ADVANCED_SYSASSERT
[    3.510348] iwlwifi 0000:02:00.0: 0x00000000 | umac branchlink1
[    3.510354] iwlwifi 0000:02:00.0: 0x8045DFC6 | umac branchlink2
[    3.510360] iwlwifi 0000:02:00.0: 0x0109112A | umac interruptlink1
[    3.510366] iwlwifi 0000:02:00.0: 0x00000000 | umac interruptlink2
[    3.510372] iwlwifi 0000:02:00.0: 0x00000005 | umac data1
[    3.510378] iwlwifi 0000:02:00.0: 0xDEADBEEF | umac data2
[    3.510384] iwlwifi 0000:02:00.0: 0xDEADBEEF | umac data3
[    3.510390] iwlwifi 0000:02:00.0: 0x00000042 | umac major
[    3.510396] iwlwifi 0000:02:00.0: 0xF1C864E0 | umac minor
[    3.510401] iwlwifi 0000:02:00.0: 0x00023213 | frame pointer
[    3.510408] iwlwifi 0000:02:00.0: 0xC0885E88 | stack pointer
[    3.510414] iwlwifi 0000:02:00.0: 0x00010C00 | last host cmd
[    3.510421] iwlwifi 0000:02:00.0: 0x00000000 | isr status reg
[    3.510439] iwlwifi 0000:02:00.0: IML/ROM dump:
[    3.510444] iwlwifi 0000:02:00.0: 0x00000B03 | IML/ROM error/state
[    3.510463] iwlwifi 0000:02:00.0: 0x00007278 | IML/ROM data1
[    3.510481] iwlwifi 0000:02:00.0: 0x00000080 | IML/ROM WFPM_AUTH_KEY_0
[    3.510494] iwlwifi 0000:02:00.0: Fseq Registers:
[    3.510503] iwlwifi 0000:02:00.0: 0x60000100 | FSEQ_ERROR_CODE
[    3.510512] iwlwifi 0000:02:00.0: 0x00440005 | FSEQ_TOP_INIT_VERSION
[    3.510522] iwlwifi 0000:02:00.0: 0x00080009 | FSEQ_CNVIO_INIT_VERSION
[    3.510533] iwlwifi 0000:02:00.0: 0x0000A652 | FSEQ_OTP_VERSION
[    3.510542] iwlwifi 0000:02:00.0: 0x00000002 | FSEQ_TOP_CONTENT_VERSION
[    3.510553] iwlwifi 0000:02:00.0: 0x4552414E | FSEQ_ALIVE_TOKEN
[    3.510562] iwlwifi 0000:02:00.0: 0x00400410 | FSEQ_CNVI_ID
[    3.510572] iwlwifi 0000:02:00.0: 0x00400410 | FSEQ_CNVR_ID
[    3.510581] iwlwifi 0000:02:00.0: 0x00400410 | CNVI_AUX_MISC_CHIP
[    3.510593] iwlwifi 0000:02:00.0: 0x00400410 | CNVR_AUX_MISC_CHIP
[    3.510605] iwlwifi 0000:02:00.0: 0x00009061 | CNVR_SCU_SD_REGS_SD_REG_DIG_DCDC_VTRIM
[    3.510619] iwlwifi 0000:02:00.0: 0x00000061 | CNVR_SCU_SD_REGS_SD_REG_ACTIVE_VDIG_MIRROR
[    3.510648] iwlwifi 0000:02:00.0: WRT: Collecting data: ini trigger 13 fired (delay=0ms).
[    4.571293] iwlwifi 0000:02:00.0: Failed to run INIT ucode: -5
[    4.583237] iwlwifi 0000:02:00.0: retry init count 1
[    4.583725] iwlwifi 0000:02:00.0: Detected Intel(R) Wi-Fi 6 AX210 160MHz, REV=0x420
[    4.735773] iwlwifi 0000:02:00.0: Microcode SW error detected. Restarting 0x0.
[    4.735914] iwlwifi 0000:02:00.0: Start IWL Error Log Dump:
[    4.735920] iwlwifi 0000:02:00.0: Transport status: 0x0000004A, valid: 6
[    4.735927] iwlwifi 0000:02:00.0: Loaded firmware version: 66.f1c864e0.0 ty-a0-gf-a0-66.ucode
[    4.735935] iwlwifi 0000:02:00.0: 0x00000071 | NMI_INTERRUPT_UMAC_FATAL
[    4.735942] iwlwifi 0000:02:00.0: 0x002002F0 | trm_hw_status0
[    4.735949] iwlwifi 0000:02:00.0: 0x00000000 | trm_hw_status1
[    4.735955] iwlwifi 0000:02:00.0: 0x004DA722 | branchlink2
[    4.735961] iwlwifi 0000:02:00.0: 0x004D0CCE | interruptlink1
[    4.735967] iwlwifi 0000:02:00.0: 0x004D0CCE | interruptlink2
[    4.735973] iwlwifi 0000:02:00.0: 0x004D94DA | data1
[    4.735979] iwlwifi 0000:02:00.0: 0x00000010 | data2
[    4.735985] iwlwifi 0000:02:00.0: 0x00000000 | data3
[    4.735991] iwlwifi 0000:02:00.0: 0x00000000 | beacon time
[    4.735997] iwlwifi 0000:02:00.0: 0x00011A97 | tsf low
[    4.736003] iwlwifi 0000:02:00.0: 0x00000000 | tsf hi
[    4.736009] iwlwifi 0000:02:00.0: 0x00000000 | time gp1
[    4.736015] iwlwifi 0000:02:00.0: 0x00023208 | time gp2
[    4.736021] iwlwifi 0000:02:00.0: 0x00000001 | uCode revision type
[    4.736027] iwlwifi 0000:02:00.0: 0x00000042 | uCode version major
[    4.736033] iwlwifi 0000:02:00.0: 0xF1C864E0 | uCode version minor
[    4.736040] iwlwifi 0000:02:00.0: 0x00000420 | hw version
[    4.736046] iwlwifi 0000:02:00.0: 0x18C89002 | board version
[    4.736052] iwlwifi 0000:02:00.0: 0x8008FF05 | hcmd
[    4.736057] iwlwifi 0000:02:00.0: 0x00020000 | isr0
[    4.736063] iwlwifi 0000:02:00.0: 0x20000000 | isr1
[    4.736069] iwlwifi 0000:02:00.0: 0x48F00002 | isr2
[    4.736075] iwlwifi 0000:02:00.0: 0x00C0000C | isr3
[    4.736081] iwlwifi 0000:02:00.0: 0x00000000 | isr4
[    4.736087] iwlwifi 0000:02:00.0: 0x00000000 | last cmd Id
[    4.736092] iwlwifi 0000:02:00.0: 0x004D94DA | wait_event
[    4.736098] iwlwifi 0000:02:00.0: 0x00000000 | l2p_control
[    4.736104] iwlwifi 0000:02:00.0: 0x00000000 | l2p_duration
[    4.736109] iwlwifi 0000:02:00.0: 0x00000000 | l2p_mhvalid
[    4.736115] iwlwifi 0000:02:00.0: 0x00000000 | l2p_addr_match
[    4.736121] iwlwifi 0000:02:00.0: 0x00000009 | lmpm_pmg_sel
[    4.736127] iwlwifi 0000:02:00.0: 0x00000000 | timestamp
[    4.736132] iwlwifi 0000:02:00.0: 0x00000024 | flow_handler
[    4.736246] iwlwifi 0000:02:00.0: Start IWL Error Log Dump:
[    4.736254] iwlwifi 0000:02:00.0: Transport status: 0x0000004A, valid: 7
[    4.736261] iwlwifi 0000:02:00.0: 0x2010070D | ADVANCED_SYSASSERT
[    4.736268] iwlwifi 0000:02:00.0: 0x00000000 | umac branchlink1
[    4.736274] iwlwifi 0000:02:00.0: 0x8045DFC6 | umac branchlink2
[    4.736281] iwlwifi 0000:02:00.0: 0x0109112A | umac interruptlink1
[    4.736287] iwlwifi 0000:02:00.0: 0x00000000 | umac interruptlink2
[    4.736293] iwlwifi 0000:02:00.0: 0x00000005 | umac data1
[    4.736299] iwlwifi 0000:02:00.0: 0xDEADBEEF | umac data2
[    4.736305] iwlwifi 0000:02:00.0: 0xDEADBEEF | umac data3
[    4.736311] iwlwifi 0000:02:00.0: 0x00000042 | umac major
[    4.736316] iwlwifi 0000:02:00.0: 0xF1C864E0 | umac minor
[    4.736322] iwlwifi 0000:02:00.0: 0x00023200 | frame pointer
[    4.736329] iwlwifi 0000:02:00.0: 0xC0885E88 | stack pointer
[    4.736334] iwlwifi 0000:02:00.0: 0x00010C00 | last host cmd
[    4.736340] iwlwifi 0000:02:00.0: 0x00000000 | isr status reg
[    4.736358] iwlwifi 0000:02:00.0: IML/ROM dump:
[    4.736364] iwlwifi 0000:02:00.0: 0x00000B03 | IML/ROM error/state
[    4.736382] iwlwifi 0000:02:00.0: 0x0000727A | IML/ROM data1
[    4.736401] iwlwifi 0000:02:00.0: 0x00000080 | IML/ROM WFPM_AUTH_KEY_0
[    4.736414] iwlwifi 0000:02:00.0: Fseq Registers:
[    4.736422] iwlwifi 0000:02:00.0: 0x60000100 | FSEQ_ERROR_CODE
[    4.736432] iwlwifi 0000:02:00.0: 0x00440005 | FSEQ_TOP_INIT_VERSION
[    4.736442] iwlwifi 0000:02:00.0: 0x00080009 | FSEQ_CNVIO_INIT_VERSION
[    4.736455] iwlwifi 0000:02:00.0: 0x0000A652 | FSEQ_OTP_VERSION
[    4.736465] iwlwifi 0000:02:00.0: 0x00000002 | FSEQ_TOP_CONTENT_VERSION
[    4.736475] iwlwifi 0000:02:00.0: 0x4552414E | FSEQ_ALIVE_TOKEN
[    4.736484] iwlwifi 0000:02:00.0: 0x00400410 | FSEQ_CNVI_ID
[    4.736503] iwlwifi 0000:02:00.0: 0x00400410 | FSEQ_CNVR_ID
[    4.736512] iwlwifi 0000:02:00.0: 0x00400410 | CNVI_AUX_MISC_CHIP
[    4.736523] iwlwifi 0000:02:00.0: 0x00400410 | CNVR_AUX_MISC_CHIP
[    4.736535] iwlwifi 0000:02:00.0: 0x00009061 | CNVR_SCU_SD_REGS_SD_REG_DIG_DCDC_VTRIM
[    4.736547] iwlwifi 0000:02:00.0: 0x00000061 | CNVR_SCU_SD_REGS_SD_REG_ACTIVE_VDIG_MIRROR
[    4.736571] iwlwifi 0000:02:00.0: WRT: Collecting data: ini trigger 13 fired (delay=0ms).
[    5.776329] iwlwifi 0000:02:00.0: Failed to run INIT ucode: -5
[    5.788240] iwlwifi 0000:02:00.0: retry init count 2

I did some more digging and found that there seems to be an issue with the current intel firmware causes issues:
https://bugs.archlinux.org/task/73387

Any way I can get write access to the host system to downgrade to version 63?

Possible related issues that have not been resolved:
 #1853
 #2026

### What operating system image do you use?

generic-x86-64 (Generic UEFI capable x86-64 systems)

### What version of Home Assistant Operating System is installed?

Home Assistant OS 9.4

### Did you upgrade the Operating System.

No

### Steps to reproduce the issue

1. Install an Intel AX210NGW
2. Boot up the machine
3. Check System->Network
4. No new Wifi Device shows up

### Anything in the Supervisor logs that might be useful for us?

```txt
No. See description
```

### Anything in the Host logs that might be useful for us?

```txt
No. See description
```

## System Information

version | core-2022.12.7
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.10.7
os_name | Linux
os_version | 5.15.80
arch | x86_64
timezone | Europe/Berlin
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 4862
Installed Version | 1.29.0
Stage | running
Available Repositories | 1207
Downloaded Repositories | 6

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | true
-- | --
subscription_expiration | 16. Januar 2023 um 01:00
relayer_connected | true
remote_enabled | false
remote_connected | false
alexa_enabled | false
google_enabled | false
remote_server | eu-central-1-4.ui.nabu.casa
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 9.4
-- | --
update_channel | stable
supervisor_version | supervisor-2022.11.2
agent_version | 1.4.1
docker_version | 20.10.19
disk_total | 109.3 GB
disk_used | 5.8 GB
healthy | true
supported | true
board | generic-x86-64
supervisor_api | ok
version_api | ok
installed_addons | Studio Code Server (5.5.0), Mosquitto broker (6.1.3), NGINX Home Assistant SSL proxy (3.2.0), Terminal & SSH (9.6.1), Check Home Assistant configuration (3.11.0), Xiaomi Mi Scale (0.2.8), Valetudo Mapper (1.11.0), Nginx Proxy Manager (0.12.3)

</details>

<details><summary>Dashboards</summary>

dashboards | 2
-- | --
resources | 3
views | 10
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | 10. November 2022 um 22:53
-- | --
current_recorder_run | 21. Dezember 2022 um 14:30
estimated_db_size | 494.54 MiB
database_engine | sqlite
database_version | 3.38.5

</details>

<details><summary>Spotify</summary>

api_endpoint_reachable | ok
-- | --

</details>

**Comment by agners** [comment]
Hm, this looks like a Linux kernel/firmware bug.

This bug report seems to match your firmware version and the issue, but the issue has been resolved and the patch is part of Home Assistant OS already:
https://bugzilla.kernel.org/show_bug.cgi?id=215523

Maybe it is related to this issue. It might be resolved in newer kernel versions:
https://bugzilla.kernel.org/show_bug.cgi?id=208425

I plan to update to 6.1 which presumably will be the next LTS kernel in January. Probably worth giving development builds  try once that is done.

**Comment by robotastic-io** [comment]
Okay, I'll wait for the january update and report back then. Got a spare stick flying around I can use in the meantime.

**Comment by dirk77** [comment]
I have exaclty the same issue with exactly the same AX210 card. Hopefully something will be done to the kernel in the near future. 
Thanks to your explanation I was able to find out that it was obviously not me that the wifi was not working ;-)

**Comment by el1-x** [comment]
I have just recently picked up one of these wireless cards for my mini PC that's running Proxmox and HAOS amongst a host of other things.

I will admit I didn't expect there to be issues with this card, but have since learned otherwise after installing and running into issues with HAOS and the Wi-Fi PCI and Bluetooth USB devices being passed through.

I am eager to see the 6.1 kernel being used for HAOS.

**Comment by el1-x** [comment]
I have just recently picked up one of these wireless cards for my mini PC that's running Proxmox and HAOS amongst a host of other things.

I will admit I didn't expect there to be issues with this card, but have since learned otherwise after installing and running into issues with HAOS and the Wi-Fi PCI and Bluetooth USB devices being passed through.

I am eager to see the 6.1 kernel being used for HAOS.

**Comment by agners** [comment]
@el1-x you can try latest dev builds from https://os-builds.home-assistant.io/ already which come with Linux 6.1:

**Comment by el1-x** [comment]
I am currently at a loss for what direction to take to get the AX210 WiFi and Bluetooth working in HA.
I am running the latest version of HAOS 10.1 using the 6.1.x kernel. HA is running as a VM under Proxmox. WiFI (PCI) and BT (USB) devices have been passed through from the host to the VM. 

Apologies for the screenshots as opposed to plain text, but without being able to SSH into HAOS this is the best I can do at the moment using the Proxmox console. This is all the information I can see via dmesg regarding both the WiFi and BT devices.

WiFi - No 'iwlwifi'' marked entries, this is all that I can find referencing the PCI device.
![image](https://user-images.githubusercontent.com/46278571/235137326-d472363b-2d0c-47df-b701-f84fe25f1238.png)

BT - FW download error recovery failed?
![image](https://user-images.githubusercontent.com/46278571/235137728-7037191d-f688-4931-b96f-4084a1d84dff.png)

**Comment by agners** [comment]
Hm, -19 means "No such device", I wonder if the device somehow disappeared again.

**Comment by el1-x** [comment]
A little more digging with dmesg | grep 00:01 reveals the below.
```
[    0.437994] pci 0000:00:1a.0: can't claim BAR 4 [io  0xd300-0xd31f]: address conflict with PCI Bus 0000:01 [io  0xd000-0xdfff]
[    0.438499] pci 0000:00:1a.1: can't claim BAR 4 [io  0xd2e0-0xd2ff]: address conflict with PCI Bus 0000:01 [io  0xd000-0xdfff]
[    0.439229] pci 0000:00:1a.2: can't claim BAR 4 [io  0xd2c0-0xd2df]: address conflict with PCI Bus 0000:01 [io  0xd000-0xdfff]
[    0.439751] pci 0000:00:1d.0: can't claim BAR 4 [io  0xd2a0-0xd2bf]: address conflict with PCI Bus 0000:01 [io  0xd000-0xdfff]
[    0.440486] pci 0000:00:1d.1: can't claim BAR 4 [io  0xd280-0xd29f]: address conflict with PCI Bus 0000:01 [io  0xd000-0xdfff]
[    0.440499] pci 0000:00:1d.2: can't claim BAR 4 [io  0xd260-0xd27f]: address conflict with PCI Bus 0000:01 [io  0xd000-0xdfff]
[    0.441302] pci 0000:00:1f.2: can't claim BAR 4 [io  0xd240-0xd25f]: address conflict with PCI Bus 0000:01 [io  0xd000-0xdfff]
[    0.441499] pci 0000:00:1f.3: can't claim BAR 4 [io  0xd200-0xd23f]: address conflict with PCI Bus 0000:01 [io  0xd000-0xdfff]
[    0.442618] pci 0000:00:01.0: vgaarb: setting as boot VGA device
[    0.442890] pci 0000:00:01.0: vgaarb: bridge control possible
[    0.443259] pci 0000:00:01.0: vgaarb: VGA device added: decodes=io+mem,owns=io+mem,locks=none
[    0.511892] pci_bus 0000:01: resource 0 [io  0xd000-0xdfff]
[    0.512281] pci_bus 0000:01: resource 1 [mem 0xc2000000-0xc21fffff]
[    0.512689] pci_bus 0000:01: resource 2 [mem 0x700000000000-0x700fffffffff 64bit pref]
```

**Comment by el1-x** [comment]
I don't believe this to be the case. I quickly fired up an Alpine Linux VM, passed though the PCI device and everything operates as expected running kernel version 5.15.109-0-lts

```
01:00.0 Network controller: Intel Corporation Wi-Fi 6 AX210/AX211/AX411 160MHz (rev 1a)
        Subsystem: Intel Corporation Wi-Fi 6 AX210 160MHz
        Physical Slot: 0
        Flags: bus master, fast devsel, latency 0, IRQ 16
        Memory at fe800000 (64-bit, non-prefetchable) [size=16K]
        Capabilities: [c8] Power Management version 3
        Capabilities: [d0] MSI: Enable- Count=1/1 Maskable- 64bit+
        Capabilities: [40] Express Endpoint, MSI 00
        Capabilities: [80] MSI-X: Enable+ Count=16 Masked-
        Capabilities: [100] Advanced Error Reporting
        Capabilities: [14c] Latency Tolerance Reporting
        Kernel driver in use: iwlwifi

[    2.227672] iwlwifi 0000:01:00.0: api flags index 2 larger than supported by driver
[    2.227707] iwlwifi 0000:01:00.0: TLV_FW_FSEQ_VERSION: FSEQ Version: 0.63.2.2
[    2.227964] iwlwifi 0000:01:00.0: loaded firmware version 66.f1c864e0.0 ty-a0-gf-a0-66.ucode op_mode iwlmvm
[    2.250611] iwlwifi 0000:01:00.0: Detected Intel(R) Wi-Fi 6 AX210 160MHz, REV=0x420
[    2.426518] iwlwifi 0000:01:00.0: loaded PNVM version 0xdbd9582f
[    2.441278] iwlwifi 0000:01:00.0: Detected RF GF, rfid=0x10d000

                                    Devices                                    
--------------------------------------------------------------------------------
  Name                  Address               Powered     Adapter     Mode      
--------------------------------------------------------------------------------
  wlan0                 xx:xx:xx:xx:xx:xx     on          phy0        station
  
  --------------------------------------------------------------------------------
  Settable  Property              Value                                          
--------------------------------------------------------------------------------
            Scanning              no                                               
            State                 connected                                        
            Connected network     BNTO                                             
            IPv6 address        
            IPv6 address               
            ConnectedBss          00:11:32:b1:81:1f                                
            Frequency             5785                                             
            Security              WPA2-Personal                                    
            RSSI                  -20 dBm                                          
            AverageRSSI           -23 dBm                                          
            TxMode                802.11ac                                         
            TxMCS                 3                                                
            TxBitrate             234000 Kbit/s                                    
            RxBitrate             6000 Kbit/s 
```

**Comment by el1-x** [comment]
> Hm, -19 means "No such device", I wonder if the device somehow disappeared again.

Thank you for this little bit of information. It pointed me in the right direction and I was able to get the Bluetooth portion of the AX210 working. 

For anyone else that may stumble across this issue, The host was 'taking control' of the Bluetooth device causing the (-19) error. The fix was to create a blacklist file within `/etc/modprobe.d/` on the Proxmox host. 

While the filename doesn't really matter, I named mine disable-bt.conf so that future me knows/remembers what I did and why. The file itself simply needs to contain the following line:
`blacklist btusb`
This will blacklist the bluetooth-usb driver from being loaded by the host. From there I was successfully able to pass-through the BT device and HA immediately detected the new device once a reboot of the VM was complete.
